src/main/java/Deadline.java

```diff
@@ -0,0 +1,21 @@
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        category = "D";
```

Comment:
If the attribute is unique to the object you might want to include this.category instead